examples/planetoid_train.py

```diff
@@ -0,0 +1,612 @@
+    gnn_choice: str,
+    model: torch.nn.Module,
+) -> torch.optim.Optimizer:
+    if gnn_choice == 'node2vec':
+        optimizer = torch.optim.SparseAdam(model.parameters(), lr=args.lr)
+    else:
+        optimizer = torch.optim.Adam(model.parameters(), lr=args.lr,
+                                     weight_decay=args.wd)
+    return optimizer
+
+
+model = get_model(args.gnn_choice).to(device)
+optimizer = get_optimizer(args.gnn_choice, model)
+if args.gnn_choice == 'node2vec':
+    num_workers = 4 if sys.platform == 'linux' else 0
+    loader = model.loader(batch_size=128, shuffle=True,
+                          num_workers=num_workers)
+
+
+def train_node2vec():
+    model.train()
+    total_loss = 0
+    for pos_rw, neg_rw in loader:
+        optimizer.zero_grad()
+        loss = model.loss(pos_rw.to(device), neg_rw.to(device))
+        loss.backward()
+        optimizer.step()
+        total_loss += loss.item()
+    return total_loss / len(loader)
+
+
+@torch.no_grad()
+def test_node2vec():
+    model.eval()
+    z = model()
+    accs = []
+    for _, mask in data('train_mask', 'val_mask', 'test_mask'):
+        acc = model.test(
+            train_z=z[data.train_mask],
+            train_y=data.y[data.train_mask],
+            test_z=z[mask],
+            test_y=data.y[mask],
+            max_iter=150,
+        )
+        accs.append(acc)
+    return accs
+
+
+def train():
+    model.train()
+    optimizer.zero_grad()
+    # forward
+    if args.gnn_choice in ['splinegnn', 'gcn']:
+        out = model(data.x, data.edge_index, data.edge_attr)
+    elif args.gnn_choice == 'gcn2':
+        out = model(data.x, data.adj_t)
+    elif args.gnn_choice == 'super_gat':
+        out, att_loss = model(data.x, data.edge_index)
+    else:
+        out = model(data.x, data.edge_index)
+    # loss
+    loss = F.cross_entropy(out[data.train_mask], data.y[data.train_mask])
+    if args.gnn_choice == 'super_gat':
+        loss += 4.0 * att_loss
+    # backward
+    loss.backward()
+    optimizer.step()
+    return loss
+
+
+@torch.no_grad()
+def test():
+    model.eval()
+    accs = []
+    # forward
+    if args.gnn_choice in ['splinegnn', 'gcn']:
+        out = model(data.x, data.edge_index, data.edge_attr)
+    elif args.gnn_choice == 'gcn2':
+        out = model(data.x, data.adj_t)
+    elif args.gnn_choice == 'super_gat':
+        out, _ = model(data.x, data.edge_index)
+    else:
+        out = model(data.x, data.edge_index)
+    # accuracy
+    for _, mask in data('train_mask', 'val_mask', 'test_mask'):
+        pred = out[mask].argmax(1)
+        acc = pred.eq(data.y[mask]).sum().item() / mask.sum().item()
+        accs.append(acc)
+    return accs
+
+
+print(f'Total time before training begins took '
+      f'{time.perf_counter() - wall_clock_start:.4f}s')
+print('Training...')
+
+# add tensorboard
+writer = SummaryWriter()
+
+times = []
+best_val_acc = test_acc = 0.
+for epoch in range(args.epochs):
+    start = time.time()
```

Comment:
nit for consistency:
```suggestion
    start = time.perf_counter()
```